packages/snaps-controllers/src/interface/utils.ts

```diff
@@ -55,13 +111,104 @@ export function assertNameIsUnique(state: InterfaceState, name: string) {
   );
 }
 
+/**
+ * Get a default asset for a given address.
+ *
+ * @param addresses - The account addresses.
+ * @param chainIds - The chain IDs to filter the assets.
+ * @param elementDataGetters - Data getters for the element.
+ * @param elementDataGetters.getAccountByAddress - A function to get an account by its address.
+ * @param elementDataGetters.getAssetsState - A function to get the MultichainAssetController state.
+ *
+ * @returns The default asset for the account or undefined if not found.
+ */
+export function getDefaultAsset(
```

Comment:
I don't entirely follow how this can return undefined? Do we not want to return some fallback if nothing is found?

Reply:
Wouldn't our other validation prevent us from getting into a state where the `undefined` value is used? If so we could simply assert instead of returning undefined

Reply:
for the `accountId` check yes, for the other one no. To check if we have assets to display I would need to add a validation step to verify that we have some assets to display but that's quite a bit of logic added for validation

Reply:
We could consider throwing then, but I'm not sure exactly how that would surface to the user if we throw here

Reply:
The accountId assertion will never be reached so good on this side.

For the other one, I would not do the check here but rather do it when we validate the interface

Reply:
But I'm not sure we should throw if we can't find any assets for a given account and chain, this is the same discussion we had last time

Reply:
If the result of the account selector is being piped into this component, doesn't that mean we would never reach a state of there being no assets? Every chain has a native asset right? Or are we worried about zero balance assets?

Reply:
You could request a network that we don't support resulting in an empty asset list but that really expensive on the controller side to check when we validate the interface, to reduce calls to a bunch of controller if you don't select a network we support we just apply the disabled state on the UI side (approved by @eriknson)

Reply:
This is the only edge case where we don't throw, if we don't have any assets to show because that would mean calling the `MultichainAssetsController` and doing some operations to determine which chains we are working with when we validate the interface which could be really expensive at a step where we want to be fast, furthermore a snap can't really determine if you have assets to show for a given network if he doesn't implement the Asset API itself making this component plausible to unexpected rejection when creating the interface.